src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -522,20 +523,22 @@ private bool ApplicableOperatorIs(MethodSymbol candidate, CSharpSyntaxNode node,
             BoundExpression operand,
             TypeSymbol operandType,
             bool hasErrors,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            bool wasSwitch = false)
         {
             bool wasExpression;
-            return BindConstantPattern(node, operand, operandType, node.Expression, hasErrors, diagnostics, out wasExpression);
+            return BindConstantPattern(node, operand, operandType, node.Expression, hasErrors, diagnostics, out wasExpression, wasSwitch);
         }
 
-        private BoundPattern BindConstantPattern(
+        internal BoundPattern BindConstantPattern(
             CSharpSyntaxNode node,
             BoundExpression left,
             TypeSymbol leftType,
             ExpressionSyntax right,
             bool hasErrors,
             DiagnosticBag diagnostics,
-            out bool wasExpression)
+            out bool wasExpression,
+            bool wasSwitch)
```

Comment:
The name of the parameter feels too vague, lots of things can be in switch. Would name isSwitchCaseLabel accurately reflect what this parameter is supposed to convey?